batch-nft-reveal/app/lib/deploy.ts

```diff
@@ -0,0 +1,34 @@
+import { ContractFactory, Signer } from 'ethers'
+
+import NFTCollection from '../artifacts/contracts/NFTCollection.sol/NFTCollection.json'
+import {
+  VRF_CALLBACK_GAS_LIMIT,
+  VRF_COORDINATOR_V2_ADDRESS,
+  VRF_GAS_LANE,
+} from '../conf/config'
+import NFTCollectionParams from './types/NFTCollectionParams'
+
+export async function deployNFTCollection(
+  nftParams: NFTCollectionParams,
+  signer: Signer,
+  chainId: number
+) {
+  const nftCollection = new ContractFactory(
+    NFTCollection.abi,
+    NFTCollection.bytecode,
+    signer
+  )
+  const deployedContract = await nftCollection.deploy(
+    nftParams.nftName,
+    nftParams.nftSymbol,
+    nftParams.nftMaxSupply,
+    nftParams.nftMintCost,
+    nftParams.nftRevealBatchSize,
+    nftParams.nftRevealInterval,
+    VRF_COORDINATOR_V2_ADDRESS[chainId],
+    nftParams.vrfSubscriptionId,
+    VRF_GAS_LANE,
+    VRF_CALLBACK_GAS_LIMIT
+  )
+  return deployedContract
+}
```

Comment:
The deployed Contract doesn't contains the transaction hash yet. We need to wait for deployment to be mined. I can wait for it in the frontend, but it's a bad style to put such logic in a frontend component. I think should be handled by the service.

Maybe you can add:
`const tx = await deployedContract.deployTransaction.wait(1)`
and return:
`return [ tx.contractAddress, tx.transactionHash]`
and the return type of the function should be `string[]`